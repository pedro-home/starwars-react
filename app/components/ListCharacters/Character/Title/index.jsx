import React from 'react';
import PropTypes from 'prop-types';

const Component = ({ text }) => {
  return <h6 className="title">{text}</h6>;
};

Component.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Component;
