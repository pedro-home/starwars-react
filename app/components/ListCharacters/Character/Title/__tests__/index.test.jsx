import React from 'react';
import { render } from '@testing-library/react';

import Title from '../index';

describe('Given <Title />', () => {
  let component;
  let defaultProps;

  beforeAll(() => {
    defaultProps = { text: 'test' };
  });

  describe('When initializes', () => {
    beforeEach(() => {
      const { container } = render(<Title {...defaultProps} />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render a text', () => {
      expect(component.querySelector('.title').innerHTML).toEqual(defaultProps.text);
    });
  });
});
