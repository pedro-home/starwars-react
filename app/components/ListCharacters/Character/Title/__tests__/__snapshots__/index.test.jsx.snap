// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Title /> When initializes Then should match the snapshot 1`] = `
<div>
  <h6
    class="title"
  >
    test
  </h6>
</div>
`;
